src/app/pages/component-sidenav/component-nav.html

```diff
@@ -10,12 +10,12 @@
       </button>
       <mat-nav-list dense [@bodyExpansion]="panel.expanded ? 'expanded' : 'collapsed'"
                     id="panel-{{category.id}}" [attr.aria-label]="category.name">
-        <a mat-list-item *ngFor="let component of category.items"
-           [routerLink]="'/' + (params | async)?.section+ '/' + component.id"
-           routerLinkActive="docs-component-viewer-sidenav-item-selected"
-           [attr.aria-current]="currentItemId === component.id ? 'page': 'false'">
+        <button mat-list-item *ngFor="let component of category.items"
```

Comment:
These should remain `<a>` elements:
* The `<a>` element signals to screen readers that this is a link and says where it goes
* You can't ctrl + click or shift + click a button
* You can't right click and copy the URL location on a button

Reply:
I agree with all of your bullet points and had those in mind when making this change.

However the mobile UX with links in the sidenav is poor because clicking a link leaves the sidenav open and the backdrop covering the page. This requires an extra tap + the knowledge of where to tap in order to see the new content. It's also possible that the user won't realize that navigation has occurred since there is no ripple, only a brief rectangular flash, the nav item text color changes, but the content is almost completely obscured behind the "over" sidenav.

With the changes to close the "over" sidenav on selection, the animation was extremely janky and ugly. Changing them to `button`s and waiting to navigate until the sidenav close Promise resolved in `selectNavItem()` was the only way that I could find to remove the janky animation.

Keeping the `<a>` tag and using `(click)="selectNavItem(component.id)"` to dismiss the "over" sidenav was janky as the navigation always started eagerly and larger page loads (like select) would jank the sidenav closing animation.

I guess that I could investigate waiting for a `NavigationEnd` event from the Router before I closed the sidenav. I'll take a look.

Reply:
Yeah, that's where my thoughts go as well

Reply:
It looks like the `NavigationEnd` router event is fired when the router is done with that lifecycle step, but not when the browser is done rendering the content. Thus it still janks on large page renders. 

It's unfortunate that we lose the ability to close the sidenav smoothly when using `routerLink` versus button `(click)` events, which give us full control of the order of operations, so that we can ensure that the sidenav closes first via `sidenav.close().then(() => doNav());`

I don't have any other ideas at this time.  For now, I think that it's better to close the sidenav on navigation, even if there is some jank, rather than forcing the mobile user to tap the backdrop to get to the content.

Reply:
Hrm. I did a production build and did some testing on Firebase Hosting (https://angularmaterial.dev/) and I'm no longer seeing any jank when using `NavigationEnd`. Perhaps it was just a side-effect of devMode.